Extension/src/telemetry.ts

```diff
@@ -92,34 +92,50 @@ export async function deactivate(): Promise<void> {
         } catch (e) {
             // Continue even if we were not able to initialize the experimentation platform.
         }
-        if (experimentationTelemetry) {
-            experimentationTelemetry.dispose();
-        }
-    }
-}
-
-export async function logDebuggerEvent(eventName: string, properties?: { [key: string]: string }): Promise<void> {
-    try {
-        await initializationPromise;
-    } catch (e) {
-        // Continue even if we were not able to initialize the experimentation platform.
     }
     if (experimentationTelemetry) {
-        const eventNamePrefix: string = "cppdbg/VS/Diagnostics/Debugger/";
-        experimentationTelemetry.sendTelemetryEvent(eventNamePrefix + eventName, properties);
+        experimentationTelemetry.dispose();
     }
 }
 
-export async function logLanguageServerEvent(eventName: string, properties?: { [key: string]: string }, metrics?: { [key: string]: number }): Promise<void> {
-    try {
-        await initializationPromise;
-    } catch (e) {
-        // Continue even if we were not able to initialize the experimentation platform.
+export function logDebuggerEvent(eventName: string, properties?: { [key: string]: string }): void {
+    const sendTelemetry = () => {
+        if (experimentationTelemetry) {
+            const eventNamePrefix: string = "cppdbg/VS/Diagnostics/Debugger/";
+            experimentationTelemetry.sendTelemetryEvent(eventNamePrefix + eventName, properties);
+        }
+    };
+
+    if (initializationPromise) {
+        try {
+            // Use 'then' instead of 'await' because telemetry should be "fire and forget".
```

Comment:
`logDebuggerEvent` and `logLanguageServerEvent` were already "fire and forget" due to not awaiting calls to it.  We don't use `no-floating-promises`.  There are many cases in which async functions are not being awaited, and cannot be awaited (such as when registered as a callback to a `registerCommand` or when registering for various VS Code events.  There are other intentional cases a well).  Are you sure you want to take this change an incur more calls to `then`?